gravitee-apim-console-webui/src/management/api/api-traffic-v4/analytics/components/analytics-layout/analytics-layout.component.ts

```diff
@@ -0,0 +1,185 @@
+
+interface BaseQueryParams {
+  from?: string;
+  to?: string;
+  period?: string;
+  [FILTER_KEYS.HTTP_STATUSES]?: string;
+  [FILTER_KEYS.PLANS]?: string;
+  [FILTER_KEYS.APPLICATIONS]?: string[];
+}
+
+@Component({
+  selector: 'analytics-layout',
+  templateUrl: './analytics-layout.component.html',
+  styleUrl: './analytics-layout.component.scss',
+})
+export class AnalyticsLayoutComponent implements OnDestroy {
+  private readonly apiId: string = this.activatedRoute.snapshot.params.apiId;
+  private activatedRouteQueryParams = toSignal(this.activatedRoute.queryParams);
+  private planService = inject(ApiPlanV2Service);
+
+  filterFields = input<string[]>([FILTER_KEYS.PLANS]);
+
+  public activeFilters: Signal<any> = computed(() => this.mapQueryParamsToFilters(this.activatedRouteQueryParams()));
+
+  apiPlans$ = this.planService
+    .list(this.activatedRoute.snapshot.params.apiId, undefined, ['PUBLISHED', 'DEPRECATED', 'CLOSED'], undefined, 1, 9999)
+    .pipe(
+      map((plans) => plans.data),
+      shareReplay(1),
+    );
+
+  constructor(
+    private readonly activatedRoute: ActivatedRoute,
+    private readonly router: Router,
+    private readonly apiAnalyticsWidgetService: ApiAnalyticsWidgetService,
+  ) {
+    effect(() => {
+      this.apiAnalyticsWidgetService.setUrlParamsData(this.mapQueryParamsToUrlParamsData(this.activatedRouteQueryParams()));
+    });
+  }
+
+  onFiltersChange(filters: any): void {
+    this.updateQueryParamsFromFilters(filters);
+  }
+
+  onRefreshFilters(): void {
+    this.apiAnalyticsWidgetService.setUrlParamsData(this.mapQueryParamsToUrlParamsData(this.activeFilters()));
+  }
+
+  ngOnDestroy(): void {
+    this.apiAnalyticsWidgetService.clearStatsCache();
+  }
+
+  private updateQueryParamsFromFilters(filters: any): void {
+    const queryParams = this.createQueryParamsFromFilters(filters);
+    this.router.navigate([], {
+      queryParams,
+      queryParamsHandling: 'replace',
+    });
+  }
+
+  private createQueryParamsFromFilters(filters: any): Record<string, any> {
+    const params: Record<string, any> = {};
+
+    if (filters.period === 'custom' && filters.from && filters.to) {
+      params.from = filters.from;
+      params.to = filters.to;
+      params.period = 'custom';
+    } else {
+      params.period = filters.period;
+    }
+
+    const fields = this.filterFields();
+    if (fields.includes(FILTER_KEYS.PLANS) && (filters as any)[FILTER_KEYS.PLANS]?.length) {
+      params[FILTER_KEYS.PLANS] = (filters as any)[FILTER_KEYS.PLANS].join(',');
+    }
+    if (fields.includes(FILTER_KEYS.HTTP_STATUSES) && (filters as any)[FILTER_KEYS.HTTP_STATUSES]?.length) {
+      params[FILTER_KEYS.HTTP_STATUSES] = (filters as any)[FILTER_KEYS.HTTP_STATUSES].join(',');
+    }
+    if (fields.includes(FILTER_KEYS.APPLICATIONS) && (filters as any)[FILTER_KEYS.APPLICATIONS]?.length) {
+      params[FILTER_KEYS.APPLICATIONS] = (filters as any)[FILTER_KEYS.APPLICATIONS].join(',');
+    }
```

Comment:
```suggestion
    fields.forEach((field) => {
       params[field] = filters[field].join(',');
    })
```
You can have `params` and `filters` support dynamic property assignment if you extend record (I think).